### PR TITLE
Audit rules for edition of /etc/shadow and /etc/gshadow 

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- New audit rules to record edition or creation of:
  - `/etc/shadow `
  - `/etc/gshadow`
- Adds CCEs to these new rules

#### Rationale:

Aligns to https://github.com/linux-audit/audit-userspace/blob/master/rules/30-ospp-v42.rules

### CODE REVIEW

**Comment by yuumasato** [comment]
@shawndwells Is something else needed when assigning CCE to a rule.

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_etc_gshadow_open/rule.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+title: 'Record Events that Modify User/Group Information via open syscall - /etc/gshadow'
+
+description: |-
+    The audit system should collect write events to /etc/gshadow file for all users and root.
+    If the <tt>auditd</tt> daemon is configured
+    to use the <tt>augenrules</tt> program to read audit rules during daemon
+    startup (the default), add the following lines to a file with suffix
+    <tt>.rules</tt> in the directory <tt>/etc/audit/rules.d</tt>:
+    <pre>-a always,exit -F arch=b32 -S open -F a1&amp;03 -F path=/etc/gshadow -F auid>={{{ auid }}} -F auid!=unset -F key=user-modify</pre>
```
Comment (jan-cerny) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_etc_gshadow_open/rule.yml]: Shouldn't there be both 32 and 64 bit?

Comment (yuumasato) [reply linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_etc_gshadow_open/rule.yml]: The 64bit audit rule is described right after, at line 17.

**Comment by redhatrises** [comment]
> @shawndwells Is something else needed when assigning CCE to a rule?

@yuumasato nope. Looks like you did it correctly.... as long as there are no duplicates. :D

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_etc_gshadow_open/rule.yml (hunk 1 above)
Comment (jan-cerny) [reply linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_etc_gshadow_open/rule.yml]: Yes, I see, but this one is for `augenrules` and on line 17 it's for `auditctl`. I thought that both rules should be set for both tools.

Comment (yuumasato) [reply linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_etc_gshadow_open/rule.yml]: I see now, well noted!

Comment (yuumasato) [reply linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_etc_gshadow_open/rule.yml]: @jan-cerny Should be fixed. I have also piggy backed same description fix for rules for `/etc/group` and `/etc/passwd`.

**Comment by yuumasato** [comment]
Rebased to fix conflicts.

**Comment by jan-cerny** [comment]
I have checked the generated RHEL8 datastream contains all 6 rules and they all contain remediations and reference the right OVAL check.

Great job! Thank you.

**Pull request merged**